feature/migration/qrcode/src/main/kotlin/app/k9mail/feature/migration/qrcode/domain/usecase/QrCodeSettingsWriter.kt

```diff
@@ -0,0 +1,44 @@
+package app.k9mail.feature.migration.qrcode.domain.usecase
+
+import android.content.Context
+import android.net.Uri
+import androidx.core.content.FileProvider
+import app.k9mail.feature.migration.qrcode.domain.QrCodeDomainContract.UseCase
+import app.k9mail.feature.migration.qrcode.domain.entity.AccountData.Account
+import app.k9mail.feature.migration.qrcode.settings.XmlSettingWriter
+import java.io.File
+
+internal class QrCodeSettingsWriter(
+    private val context: Context,
+    private val xmlSettingWriter: XmlSettingWriter,
+) : UseCase.QrCodeSettingsWriter {
+    override fun write(accounts: List<Account>): Uri {
+        val file = getSettingsFile()
+        writeSettingsToFile(file, accounts)
+
+        val authority = "${context.packageName}.qrcode.settings"
+        return FileProvider.getUriForFile(context, authority, file)
```

Comment:
The file is created and overwritten correctly when importing new data. Nevertheless, it persists on the device until the app is removed or a new QR code is successfully scanned. I believe it should be automatically deleted after a successful import.